### DISCUSSION TITLE
How to be informed when new strings need translation?

### DISCUSSION THREAD

**Discussion by Patriccollu** [aclap-dev/video-downloadhelper#148] [Language & Translation]
Hello,

I'm Corsican translator for Video DownloadHelper and I try to maintain the translation up-to-date as regularly as possible.

In the past I was informed, thanks to the [VDH Internationalization Google group](https://groups.google.com/g/video-downloadhelper-internationalization), when another translator had updated his/her locale but it wasn't an evidence that there was new strings that need translation! Of course, I installed on Firefox the beta version of the VDH add-on and check from time to time the `Missing strings` on `Settings > Translation` but this is a manual procedure.

The translators need an automated way to be informed when there are new or modified strings that need translation.

If the template `messages.json` - that contains all the strings in English - could be stored in a specific place on GitHub, it would be easy to monitor the GitHub's Atom feed to be informed when there was a change (_commit_) made to that file.

What do you think of this proposal?

Thanks,
Patriccollu.

**Comment by paulrouget** [comment]
Oh! I love the idea a lot! I'll work on that :)

**Comment by Patriccollu** [comment]
Hello @paulrouget 
Any news on this request?
Thanks,
Patriccollu.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

When I go into the VDH settings, navigate to the `More settings` page, & scroll it down to the bottom, I see this:

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6ac5deb7-d7c4-4aee-9d35-8ce8440819a2)

Would that not serve your purposes?

**Comment by Patriccollu** [comment]
Hi @GrampaWildWilly 

Thanks for your help but this does not _really_ cover my needs. What your screenshot shows is the New UI's equivalent to the v8's manual procedure explained in my comment: the need to check from time to time the `Missing strings` on `Settings > Translation`. Additionnally, this can only be done when the new version is released... and installed!

Actually the translators who want to be proactive need a _way to be automatically informed_, such as receiving an email, as soon as the English `messages.json` template file is updated - for example to add new strings - and _before a new version is built and released_. This allows the translators to translate the new strings and deliver the new localization to the VDH team in order to be included in the next release.

If the English `messages.json` template file is stored on a GitHub repository, you can easily implement a procedure to automatically receive an email by monitoring the Atom feed on the file commits. The History function also shows what strings were added, changed, or deleted since the previous version.

Thanks again,
Patriccollu.

**Comment by paulrouget** [comment]
@Patriccollu https://github.com/aclap-dev/vdhlocales :)

**Comment by Patriccollu** [comment]
@paulrouget 
Hi Paul,

This looks to cover more than my original need!

Let me check if I understand correctly:
1. `en_US.toml` in the above repository is the English reference file which is updated every time strings are added, modified, or deleted.
2. As you accept Pull Requests on [vdhlocales repository](https://github.com/aclap-dev/vdhlocales#readme), I can submit a PR to propose changes on `co.toml`.
3. The [Corsican local discussion thread](https://github.com/aclap-dev/video-downloadhelper/discussions/134) may no longer be used.

Please can you just confirm these three points?

Best regards,
Patriccollu.

**Reply by paulrouget** [reply]
> 1. `en_US.toml` in the above repository is the English reference file which is updated every time strings are added, modified, or deleted.

Yes.

> 2. As you accept Pull Requests on [vdhlocales repository](https://github.com/aclap-dev/vdhlocales#readme), I can submit a PR to propose changes on `co.toml`.

Yes.

> 3. The [Corsican local discussion thread](https://github.com/aclap-dev/video-downloadhelper/discussions/134) may no longer be used.

Yes.

**Comment by Patriccollu** [comment]
@paulrouget 
Great! I'm going to close this discussion!

For your information, and before updating `co.toml`, I've raised a new question in discussion #896 regarding _string format and delimiters in .toml localization files_.

Thanks again,
Patriccollu.
